randovania/gui/main_window.py

```diff
@@ -321,6 +322,20 @@ def highlight_logo(
         self.main_tab_widget.setCurrentIndex(0)
 
     def closeEvent(self, event):
+        if self.has_background_process:
+            event.ignore()
+            dialog = QtWidgets.QMessageBox(
```

Comment:
use async_dialog.warning instead

Reply:
I was avoiding trying to mix async with non async code, since I'm not sure how to call async code from non async funciton

using something like `await async_dialog.warning` in closeEvent gives me 
"SyntaxError: 'await' outside async function" errors

Reply:
I wanted to avoid code copy paste but looks like we're already doing this sort of thing a few times in the codebase.
avoiding this means doing some hacky shenanigans with asyncio so probably not worth it.

Reply:
`closeEvent` in particular must not be async, because we need to blockingly handle the event